src/transformers/models/mistral3/modular_mistral3.py

```diff
@@ -0,0 +1,451 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch import nn
+
+from ...activations import ACT2FN
+from ...image_processing_utils import BatchFeature
+from ...image_utils import ImageInput, load_image
+from ...processing_utils import Unpack, _validate_images_text_input_order
+from ...tokenization_utils_base import PreTokenizedInput, TextInput
+from ...utils import is_torchdynamo_compiling, logging
+from ..llava.modeling_llava import LlavaCausalLMOutputWithPast, LlavaForConditionalGeneration
+from ..mistral.modeling_mistral import MistralRMSNorm
+from ..pixtral.processing_pixtral import PixtralProcessor, PixtralProcessorKwargs, is_image_or_image_url
+from .configuration_mistral3 import Mistral3Config
+
+
+logger = logging.get_logger(__name__)
+
+
+class Mistral3ProcessorKwargs(PixtralProcessorKwargs):
+    pass
+
+
+class Mistral3Processor(PixtralProcessor):
+    r"""
+    Constructs a Mistral3 processor which wraps a Mistral3 image processor and a Mistral3 tokenizer into a single processor.
+
+    [`Mistral3Processor`] offers all the functionalities of [`PixtralImageProcessor`] and [`LlamaTokenizerFast`]. See the
+    [`~Mistral3Processor.__call__`] and [`~Mistral3Processor.decode`] for more information.
+
+    Args:
+        image_processor ([`PixtralImageProcessor`], *optional*):
+            The image processor is a required input.
+        tokenizer ([`LlamaTokenizerFast`], *optional*):
+            The tokenizer is a required input.
+        patch_size (`int`, *optional*, defaults to 14):
+            Patch size from the vision tower.
+        chat_template (`str`, *optional*): A Jinja template which will be used to convert lists of messages
+            in a chat into a tokenizable string.
+        spatial_merge_size (`int`, *optional*, defaults to 2):
+            The downsampling factor for the spatial merge operation.
+        image_token (`str`, *optional*, defaults to `"[IMG]"`):
+            Special token used to denote image location.
+        image_break_token (`str`, *optional*, defaults to `"[IMG_BREAK]"`):
+            Special token used to denote the end of a line of pixels in an image.
+        image_end_token (`str`, *optional*, defaults to `"[IMG_END]"`):
+            Special token used to denote the end of an image input.
+    """
+
+    valid_kwargs = [
+        "chat_template",
+        "patch_size",
+        "image_token",
+        "image_break_token",
+        "image_end_token",
+        "spatial_merge_size",
+    ]
+
+    def __init__(
```

Comment:
shouldn't the max image size be defined here somewhere? It's 1540 in the case of mistral 3.1 - small. Which means that every image that's bigger than 1540 will be downsampled to 1540